app.py

```diff
@@ -1360,7 +1360,18 @@ def get_admin_classes_page(user):
     if not is_admin(user):
         return utils.error_page(error=403, ui_message='unauthorized')
 
-    classes = DATABASE.all_classes()
+    classes = []
+    for Class in DATABASE.all_classes():
+        current_class = {
+            "name": Class.get('name'),
+            "teacher": Class.get('teacher'),
+            "students": len(Class.get('students')) if 'students' in Class else 0,
+            "id": Class.get('id'),
+            "last_used": utils.datetotimeordate(utils.mstoisostring(DATABASE.user_by_username(Class.get('teacher')).get('last_login')))
```

Comment:
I'm not too familiar with dynamo / performance / load and how many classes there are currently - would it perhaps be a bit better to include `last_teacher_login` as part of the `Class` *class* (😎) and update this on teacher activity?

Reply:
Good idea! We currently don't support that many classes; just looked at the database: 45 at the moment. I agree that from a performance perspective it is a good idea to add the `last_log` to the class data. But I want to prevent storing duplicate data in too many places. Especially because of the small scale I think this works fine. The data can only be retrieved by the admin, so we don't have to worry about database load too much. However, if we scale to more classes we maybe don't have a choice to change but for know I suggest to keep it this way. (Will add an inline comment though).

Reply:
Good suggestion @ThomasKluiters and good idea to keep an eye on this! For now I am the only admin so indeed it is probably fine but it might get out of hand one day :)